utils/wasm-gen/src/generator/syscalls/imports.rs

```diff
@@ -242,36 +242,92 @@ impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
 }
 
 impl<'a, 'b> SysCallsImportsGenerator<'a, 'b> {
-    /// Generates a function which calls "properly" the `gr_reservation_send`.
-    fn generate_send_from_reservation(&mut self) {
-        let (reserve_gas_idx, reservation_send_idx) = {
-            let maybe_reserve_gas = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReserveGas))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-            let maybe_reservation_send = self
-                .sys_calls_imports
-                .get(&InvocableSysCall::Loose(SysCallName::ReservationSend))
-                .map(|&(_, call_indexes_handle)| call_indexes_handle);
-
-            match maybe_reserve_gas.zip(maybe_reservation_send) {
-                Some(indexes) => indexes,
+    /// Returns the indexes of invocable sys-calls.
+    fn invocable_sys_calls_indexes<const N: usize>(
+        &self,
+        sys_call: SysCallName,
```

Comment:
btw, if you still don't want `sys_call` be in the signature of this method, you can remove that, but need to log name of the precise sys-call invocation being built (in generate_send_from_reservation, for example)

Reply:
On the other hand, it's easier to understand what we want to register as precise sys-call if we write it at the beginning of the function.

Reply:
But the only thing this `fn` does is mapping sys_calls[] <-> import_idx[] so ideally it shouldn't contain any other logics.